### PR TITLE
Fix ContentType.Alias matching in PublishedContentExtensions and PublishedElementExtensions

### LINKED ISSUE #6571: ChildrenOfType (and related methods) look(s) at substring of the alias
When using the method `ChildrenOfType`, I expect only children to be matched whenever the alias of the page matches exactly with the requested alias. Instead, the code also returns matches where the alias of the page is a substring of the requested alias.

### Steps to reproduce

* Make the following structure in Umbraco:
- Home (PageHome)
    - About Us (PageContent)
    - Some News (PageContentNews)
* On the PageHome template, call `var newsPage = Model.FirstChildOfType("PageContentNews")`
* Expected: 'Some News' is returned.
* Actual: 'About Us' is returned.

### Expected result
Page 'Some News' is returned.

### Actual result
Page 'About Us' is returned.

### Specifics

The bug is caused by the fact that the code calls `contentTypeAlias.InvariantContains(x.ContentType.Alias)`, instead of simply `contentTypeAlias == x.ContextType.Alias`. This bug exists in the newest version of the v8/dev branch, and affects at least all 8.1.x and 8.2.x versions:

https://github.com/umbraco/Umbraco-CMS/blob/1d94c4b9bbf026cc313bb04a561862ffe53fb033/src/Umbraco.Web/PublishedContentExtensions.cs#L886-L896

I suspect the bug was introduced because the signature of the method changed somewhere, as `string alias` was `params string[] alias` before. Both a string and an array have a `Contains` method, but they obviously mean something different.. If you confirm that this is indeed a bug, I will try to see if I can put a PR together.

### PR DESCRIPTION
**Pull request by rbottema**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

Fixes #6571.

### Description

I went through PublishedContentExtensions and PublishedElementExtensions, fixing all occurrences of Content.Type.Alias matching to what is described in #6571.

The first commit fixes the bug (using Contains instead of Equals) for the ChildrenOfType method (this was the only method with the bug). The second commit changes all matching on ContentType.Alias to use invariant matching.

Locally, all tests (that run) pass. I looked into created a test for the bug case to prevent regression, but I could not quickly figure out how to do so properly.

To test the changes: see reproduction steps in #6571.

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Thanks @rbottema - we'll have a look soon!

File: src/Umbraco.Web/PublishedContentExtensions.cs [hunk 1]
```diff
@@ -892,7 +892,7 @@ public static IEnumerable<IPublishedContent> Children(this IPublishedContent con
         /// <returns>The children of the content, of any of the specified types.</returns>
         public static IEnumerable<IPublishedContent> ChildrenOfType(this IPublishedContent content, string contentTypeAlias, string culture = null)
         {
-            return content.Children(x => contentTypeAlias.InvariantContains(x.ContentType.Alias), culture);
+            return content.Children(x => contentTypeAlias.InvariantEquals(x.ContentType.Alias), culture);
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/PublishedContentExtensions.cs]: As `x.ContentType.Alias` shouldn't ever be `null`, it's better to swap this to:
```suggestion
            return content.Children(x => x.ContentType.Alias.InvariantEquals(contentTypeAlias), culture);
```
This also keeps it in the same order as the other methods 👍

File: src/Umbraco.Web/PublishedElementExtensions.cs [hunk 2]
```diff
@@ -165,8 +165,7 @@ public static IndexedArrayItem<TContent>[] ToIndexedArray<TContent>(this IEnumer
         public static IEnumerable<T> OfTypes<T>(this IEnumerable<T> contents, params string[] types)
             where T : IPublishedElement
         {
-            types = types.Select(x => x.ToLowerInvariant()).ToArray();
-            return contents.Where(x => types.Contains(x.ContentType.Alias.ToLowerInvariant()));
+            return contents.Where(x => types.InvariantContains(x.ContentType.Alias));
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/PublishedElementExtensions.cs]: Should we validate and short-circuit if `types` is `null` or an empty array?

```suggestion
            if (types == null || types.Length == 0) return Enumerable.Empty<T>();
            
            return contents.Where(x => types.InvariantContains(x.ContentType.Alias));
```

**Comment by nul800sebastiaan** [comment]
Thanks @rbottema for this excellent update and figuring out the root cause of this bug, well done!

Congrats on your first PR merged for Umbraco CMS, great work! 👍 
I've added a shiny new contributor badge to your Our Umbraco profile: https://our.umbraco.com/members/id:275392/ 🏅

**Comment by nul800sebastiaan** [comment]
Thanks also @ronaldbarendse for the feedback, good stuff!

**Comment by nul800sebastiaan** [comment]
Hi there @rbottema,

First of all: **A big #H5YR** for contributing to Umbraco during Hacktoberfest! We are very thankful for the huge amount of PRs submitted, and all the amazing work you've been doing 🥇 

We have had another record breaking amount of PR's in both the CMS, Documentation, and other smaller public repositories. You can read all about the numbers in the [Umbraco Hacktoberfest round-up blog post](https://umbraco.com/blog/pr-team-update-november-2019-hacktoberfest-round-up/).

Due to the amazing work you and others in the community have been doing, we've had a bit of a hard time keeping up. 😅 While all of the PRs for Hacktoberfest might not have been merged yet, you still qualify for receiving some Umbraco swag, congratulations! 🎉 

In the spirit of Hacktoberfest we've prepared some **exclusive Umbraco swag** for all our contributors - _including you_! 

[Receive your swag!](https://docs.google.com/forms/d/e/1FAIpQLSc-K97Zl5k86cVebKWWqieLM4Fm_jCMGVdbQdYnq4IUJ7EvZg/viewform) :point_left: Please follow this link to fill out and submit the form, **before December 19th 2019**.

Following this date we'll be sending out all the swag, which also means that it might not reach your doorstep before February, so please bear with us and be patient :pray:

The only thing left to say is **thank you so much for participating in Hacktoberfest**! We really appreciate the help!

Kind regards,
The Umbraco PR Team & The Documentation Curators

**Pull request merged**